packages/yourloops/test/integration/pages/login/login-page.spec.tsx

```diff
@@ -0,0 +1,112 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import * as auth0Mock from '@auth0/auth0-react'
+import { act, screen, within } from '@testing-library/react'
+import { checkFooter } from '../../assert/footer'
+import i18n from 'i18next'
+import { renderPage } from '../../utils/render'
+import userEvent from '@testing-library/user-event'
+
+describe('Login page', () => {
+  const loginWithRedirectMock = jest.fn()
+  beforeAll(() => {
+    (auth0Mock.useAuth0 as jest.Mock).mockReturnValue({
+      isAuthenticated: false,
+      isLoading: false,
+      user: undefined,
+      loginWithRedirect: loginWithRedirectMock
+    })
+  })
+
+  it('should render entire page with correct elements', async () => {
+    renderPage('/')
+
+    const header = screen.getByTestId('login-page-header')
+    const registerButton = within(header).getByRole('button', { name: 'Register' })
+    const loginButton = within(header).getByRole('button', { name: 'Connection' })
+    const contactLink = within(header).getByRole('link', { name: 'Contact' })
+    const infoContainer = screen.getByTestId('info-container')
+    const moreInfoLink = within(infoContainer).getByRole('link')
+
+    // Images
+    expect(screen.getByAltText('Yourloops brand colors')).toBeVisible()
+    expect(screen.getByAltText('Yourloops laptop')).toBeVisible()
+
+    // header
+    expect(header).toBeVisible()
+    expect(within(header).getByTestId('header-main-logo')).toBeVisible()
+    expect(registerButton).toBeVisible()
+    expect(loginButton).toBeVisible()
+    expect(contactLink).toBeVisible()
+    expect(contactLink).toHaveAttribute('href', 'mailto:yourloops@diabeloop.com')
+    expect(within(header).getByTestId('language-icon')).toBeVisible()
+    expect(within(header).getByTestId('language-selector')).toBeVisible()
+
+    // main content
+    expect(within(infoContainer).getByText('is a web application offered by Diabeloop in order to facilitate the monitoring of patients with diabetes using compatible medical devices.')).toBeVisible()
+    expect(within(infoContainer).getByText('It provides access to visual representations of data that are automatically uploaded from DBL devices: blood sugar levels, insulin, carbohydrates and physical activity.')).toBeVisible()
+    expect(within(infoContainer).getByText('YourLoops also enables patients to securely share their data with their medical team or caregivers.')).toBeVisible()
+    expect(within(infoContainer).getByText('For health professionals, YourLoops is the platform for monitoring their patients. Through these features, it saves time and facilitates therapeutic decision-making.')).toBeVisible()
+    expect(moreInfoLink).toBeVisible()
+    expect(moreInfoLink).toHaveAttribute('href', 'https://www.dbl-diabetes.com')
+    checkFooter({ needFooterLanguageSelector: false })
+
+    // More info link should disappear if language is french
+    await act(async () => {
+      await i18n.changeLanguage('fr')
```

Comment:
Maybe change the lang by using the language selector in the header? It would be better and closer to a human interaction